StackExchange.Redis/StackExchange/Redis/RedisDatabase.cs

```diff
@@ -45,6 +47,134 @@ public RedisValue DebugObject(RedisKey key, CommandFlags flags = CommandFlags.No
+            return ExecuteSync(msg, ResultProcessor.RedisValue);
+        }
+
+        public string[] GeoHash(RedisKey key, string[] members, CommandFlags flags = CommandFlags.None)
+        {
+            if(members == null)throw new ArgumentNullException(nameof(members));
+            var redisValues = new RedisValue[members.Length];
+            for (var i = 0; i < members.Length; i++) redisValues[i] = members[i];
+            var msg = Message.Create(Database, flags, RedisCommand.GEOHASH, key, redisValues);
+            var results = ExecuteSync(msg, ResultProcessor.RedisValueArray);
+            var retArray = new string[results.Length];
+            for (int i = 0; i < results.Length; i++)
+            {
+                retArray[i] = (string)results[i];
+            }
+            return retArray;
+        }
+
+        public string[] GeoHash(RedisKey key, string member, CommandFlags flags = CommandFlags.None)
+        {
+            return GeoHash(key, new[] {member}, flags);
+        }
+
+        public GeoPosition?[] GeoPos(RedisKey key, string[] members, CommandFlags flags = CommandFlags.None)
+        {
+            if (members == null) throw new ArgumentNullException(nameof(members));
+            var redisValues = new RedisValue[members.Length];
+            for (var i = 0; i < members.Length; i++) redisValues[i] = members[i];
+            var msg = Message.Create(Database, flags, RedisCommand.GEOPOS, key, redisValues);
+            return ExecuteSync(msg, ResultProcessor.RedisGeoPosition);
+        }
+
+        public GeoPosition? GeoPos(RedisKey key, string member, CommandFlags flags = CommandFlags.None)
+        {
+            return GeoPos(key, new string[] {member}, flags).FirstOrDefault();
+        }
+
+        public GeoRadiusResult[] GeoRadius(RedisKey key, GeoRadius geoRadius,CommandFlags flags = CommandFlags.None)
+        {
+            var redisValues = new List<RedisValue>();
+            redisValues.Add(geoRadius.geoPosition.longitude);
+            redisValues.Add(geoRadius.geoPosition.latitude);
+            redisValues.Add(geoRadius.radius);
+            redisValues.Add(_redisUnits[(int)geoRadius.geoUnit]);
+            if(geoRadius.geoRadiusOptions.HasFlag(GeoRadiusOptions.WITHCOORD))
+                redisValues.Add(GeoRadiusOptions.WITHCOORD.ToString());
+            if (geoRadius.geoRadiusOptions.HasFlag(GeoRadiusOptions.WITHDIST))
+                redisValues.Add(GeoRadiusOptions.WITHDIST.ToString());
+            if (geoRadius.geoRadiusOptions.HasFlag(GeoRadiusOptions.WITHHASH))
+                redisValues.Add(GeoRadiusOptions.WITHHASH.ToString());
+            if(geoRadius.maxReturnCount >0)
+                redisValues.Add(geoRadius.maxReturnCount);
+            var msg = Message.Create(Database, flags, RedisCommand.GEORADIUS, key, redisValues.ToArray());
+            var items = ExecuteSync(msg, ResultProcessor.RedisGeoRadius);
+
+            var arr = new GeoRadiusResult[items.Length];

```

Comment:
it looks to me like a lot of this should be moved to the processor - in particular, this would avoid having to duplicate it (or refactor it) for the `async` version later